src/main/java/io/anlessini/SearchLambda.java

```diff
@@ -27,72 +31,68 @@
+  private static final Logger LOG = LogManager.getLogger(SearchLambda.class);
 
-public class SearchLambda {
-  private static final String INDEX_BUCKET = "INDEX_BUCKET";
-  private static final String INDEX_KEY = "INDEX_KEY";
+  private final S3Directory directory;
+  private final IndexReader reader;
+  private static final String S3_INDEX_BUCKET = System.getenv("INDEX_BUCKET");
+  private static final String S3_INDEX_KEY = System.getenv("INDEX_KEY");
 
-  private static S3Directory directory = null;
-  private static IndexReader reader = null;
+  private final DynamoDB dynamoDB;
+  private final String DYNAMODB_TABLE_NAME = System.getenv("TABLE_NAME");
+  private final Table dynamoTable;
 
-  private String bucket;
-  private String key;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static final ObjectWriter writer = objectMapper.writer();
 
-  public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context)
-          throws IOException {
+  public SearchLambda() throws IOException {
+    directory = new S3Directory(S3_INDEX_BUCKET, S3_INDEX_KEY);
+    reader = DirectoryReader.open(directory);
 
-    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
-    StringBuilder sb = new StringBuilder();
-    String line;
-    while ((line = reader.readLine()) != null) {
-      sb.append(line);
-    }
-    JSONObject json = new JSONObject(sb.toString());
-
-
-    ObjectMapper mapper = new ObjectMapper();
-    LambdaLogger logger = context.getLogger();
+    // initializing the dynamodb instance
+    dynamoDB = new DynamoDB(AmazonDynamoDBClientBuilder.defaultClient());
+    dynamoTable = dynamoDB.getTable(DYNAMODB_TABLE_NAME);
+  }
 
-    logger.log("Input: \n");
-    String qstring = json.getJSONObject("queryStringParameters").get("query").toString();
+  @Override
+  public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent input, Context context) {
+    APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent();
+    String queryString = input.getQueryStringParameters().get("query");
+
+    try {
+      String hits = search(queryString);
+      response.setStatusCode(200);
+      response.setBody(hits);
+    } catch (IOException e) {
+      LOG.error(e.getMessage(), e);
+      response.setStatusCode(500);
+      try {
+        response.setBody(writer.writeValueAsString(Map.of("message", e.getMessage())));
+      } catch (JsonProcessingException ex) {
+        LOG.error("Error writing JSON message", ex);
+      }
+    }
 
-    String hits = search(qstring, context);
-    outputStream.write(("{\"statusCode\": 200, \"headers\": { \"Access-Control-Allow-Origin\": \"*\" }," +
-            "\"body\": " + mapper.writeValueAsString(hits) + "}").getBytes());
+    response.setHeaders(Map.of(
+        "Content-Type", "application/json"
+    ));
+    return response;
```

Comment:
Does the structure of the json that is being returned change at all? It's fine if it does, but some stuff might have to be changed on the anlessini-demo repo if it's different.

Reply:
The response structure is mostly the same, with the only change being the headers. I removed `Access-Control-Allow-Origin` as I find it irrelevant in our scenario, and added `Content-Type=application/json` to indicate the body is JSON text.

Reply:
Ah okay thats good then